src/tools/idlc/include/libidlc/idl_defs.h

```diff
@@ -8,25 +8,38 @@
 //
 // SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
 
-#ifndef IDLC_GENERATOR_H
-#define IDLC_GENERATOR_H
+#ifndef IDL_DEFS_H
```

Comment:
The `idl_defs.h` header seems to contain only generic definition, so should it be part of `idlc` and not `libidlc`?

Reply:
Moved this into a common generator library which contains the functions used by all idl generators, like names, namespaces, type descriptors etc.
Now other generators (idlcxx) no longer have to link against libidlc